### ISSUE TITLE
GitHub dependency graph

### ISSUE THREAD

**Issue by Alberthaff** [badges/shields#1550]
It would be great to have a shield displaying stats from GitHub's dependency graph (dependent repos / packages):

https://github.com/badges/shields/network/dependents

**Comment by chris48s** [comment]
Thanks for the suggestion. Our data for the GitHub badges comes from their API. I don't think GitHub surface this information via an API endpoint yet, do they?

**Comment by Alberthaff** [comment]
Yes, you are right.
Unfortunately, their API doesn't seem to have an endpoint to retrieve the dependency graph.

**Comment by paulmelnikow** [comment]
Is there another service that could provide this? For example, we use https://libraries.io/ for bower. Would they have a relevant endpoint?

**Comment by chris48s** [comment]
If the project is a package indexed by libraries.io, the [project-dependencies](https://libraries.io/api/#project-dependencies), [project-dependents](https://libraries.io/api/#project-dependents) (these are both about other _packages_ that your package depends on, or is dependent on) and the [dependent-repositories](https://libraries.io/api/#project-dependent-repositories) endpoint (which is about _repositories_ that depend on your package) are all relevant in this problem domain but won't produce the same numbers as the GH dependency graph. See https://github.com/librariesio/libraries.io/issues/2005

**Comment by paulmelnikow** [comment]
This came up again in #1858, where @sindresorhus pointed out there is a GraphQL endpoint for the dependency graph: https://developer.github.com/changes/2018-04-24-preview-dependency-graph-and-vulnerability-hooks/

However:

> That endpoint doesn't seem to provide access to the dependents number, only the dependencies number.
> 
> IIRC scraping github violates their ToS so I'm not sure the best way forward here.

As @chris48s points out above, Libraries.io has a different number available, but it doesn't match Github. I added a badge for that in #1895.

**Comment by chris48s** [comment]
If anyone is interested in working on this, there is now a `GithubAuthV4Service` base class for working with the Github GraphQL API. https://github.com/badges/shields/blob/master/services/github/github-forks.service.js is an example of another bad that uses it.

**Comment by nedbat** [comment]
I haven't tried it, but does this GraphQL object provide the data needed?  https://docs.github.com/en/graphql/reference/objects#dependencygraphmanifest

GitHub is now displaying the number of dependents on the front page of a repo as "Used by", which will make more people want a badge for it:
<img width="587" alt="image" src="https://user-images.githubusercontent.com/23789/198274769-78f67c6c-53a9-4a04-a33f-828b21f53af6.png">

**Comment by chris48s** [comment]
Yes. As noted in the link 2 posts up ( https://developer.github.com/changes/2018-04-24-preview-dependency-graph-and-vulnerability-hooks/ ), `dependencyGraphManifests` would be the right query if anyone wants to work on this. Interestingly, 4 years on that query still requires passing a preview header to enable it: `application/vnd.github.hawkgirl-preview+json` refs https://docs.github.com/en/graphql/overview/schema-previews#access-to-a-repositories-dependency-graph-preview

**Comment by mrf345** [comment]
It seems both the REST and GraphQL APIs don't expose the dependents (Used-by) yet. So I've created a small temporary wrapper in case someone's else wants to use it https://github.com/mrf345/gh-used-by
